libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2341,6 +2347,47 @@ bool TypeChecker::visit(IndexAccess const& _access)
 	return false;
 }
 
+bool TypeChecker::visit(IndexRangeAccess const& _access)
+{
+	_access.baseExpression().accept(*this);
+	TypePointer exprType = type(_access.baseExpression());
+	ArrayType const* arrayType = nullptr;
+
+	Expression const& start = _access.startExpression();
+	Expression const* end = _access.endExpression();
+
+	bool isLValue = false;
```

Comment:
Do you need these two local variables?

Reply:
``isLValue`` not exactly so far - but I kept it as a reminder that this has to be considered when implementing memory and storage slices... and ``isPure`` is assigned a few times and just way shorter than ``_access.baseExpression().annotation().isPure`` - it's done similarly in ``IndexAccess``